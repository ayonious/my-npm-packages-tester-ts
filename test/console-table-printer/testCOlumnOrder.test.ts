import { expect } from '@jest/globals';
import { Table } from 'console-table-printer';

describe('Example: Print a simple Table without table instance creation', () => {
  it(`simple print`, () => {
    //Create a table
    const p = new Table({
      columns: [
        { name: 'eighty' },
        { name: 'index', alignment: 'left', color: 'yellow' },
        { name: 'text', alignment: 'right' },
        { name: 'third' },
        { name: 'fourth' },
        { name: 'twelve' },
        { name: 'fifth' },
        { name: 'sixth' },
        { name: 'seventh' },
      ],
    });

    p.addRow({ index: 1, text: 'red wine', value: 10.212 }, { color: 'green' }); // row coloring

    p.addRow({ index: 2, text: 'green gemuse', value: 20.0 });
    p.addRow(
      {
        text: 'gelb bananen',
        value: 100,
        third: 3,
        fifth: 7,
        is_priority_today: 'Y',
        fourth: 4,
      },
      { color: 'yellow' }
    );
    p.addRow(
      {
        sixth: 6,
        seventh: 7,
        index: 3,
        text: 'rosa hemd wie immer',
        value: 100,
      },
      { color: 'cyan' }
    );
    p.addRow(
      {
        sixth: 6,
        seventh: 7,
        twelve: 3,
        eighty: 'rosa hemd wie immer',
      },
      { color: 'green' }
    );
    expect(p.render()).toMatchSnapshot();
  });
});
