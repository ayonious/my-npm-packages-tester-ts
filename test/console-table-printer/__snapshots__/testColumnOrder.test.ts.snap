// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Example: Print a simple Table without table instance creation simple print 1`] = `
"┌─────────────────────┬───────┬─────────────────────┬───────┬────────┬────────┬───────┬───────┬─────────┬────────┬───────────────────┐
│[37m [0m[01m             eighty[0m │[37m [0m[01mindex[0m │[37m [0m[01m               text[0m │[37m [0m[01mthird[0m │[37m [0m[01mfourth[0m │[37m [0m[01mtwelve[0m │[37m [0m[01mfifth[0m │[37m [0m[01msixth[0m │[37m [0m[01mseventh[0m │[37m [0m[01m value[0m │[37m [0m[01mis_priority_today[0m │
├─────────────────────┼───────┼─────────────────────┼───────┼────────┼────────┼───────┼───────┼─────────┼────────┼───────────────────┤
│[37m [0m[32m                   [0m │[37m [0m[33m1    [0m │[37m [0m[32m           red wine[0m │[37m [0m[32m     [0m │[37m [0m[32m      [0m │[37m [0m[32m      [0m │[37m [0m[32m     [0m │[37m [0m[32m     [0m │[37m [0m[32m       [0m │[37m [0m[32m10.212[0m │[37m [0m[32m                 [0m │
│[37m [0m[37m                   [0m │[37m [0m[33m2    [0m │[37m [0m[37m       green gemuse[0m │[37m [0m[37m     [0m │[37m [0m[37m      [0m │[37m [0m[37m      [0m │[37m [0m[37m     [0m │[37m [0m[37m     [0m │[37m [0m[37m       [0m │[37m [0m[37m    20[0m │[37m [0m[37m                 [0m │
│[37m [0m[33m                   [0m │[37m [0m[33m     [0m │[37m [0m[33m       gelb bananen[0m │[37m [0m[33m    3[0m │[37m [0m[33m     4[0m │[37m [0m[33m      [0m │[37m [0m[33m    7[0m │[37m [0m[33m     [0m │[37m [0m[33m       [0m │[37m [0m[33m   100[0m │[37m [0m[33m                Y[0m │
│[37m [0m[36m                   [0m │[37m [0m[33m3    [0m │[37m [0m[36mrosa hemd wie immer[0m │[37m [0m[36m     [0m │[37m [0m[36m      [0m │[37m [0m[36m      [0m │[37m [0m[36m     [0m │[37m [0m[36m    6[0m │[37m [0m[36m      7[0m │[37m [0m[36m   100[0m │[37m [0m[36m                 [0m │
│[37m [0m[32mrosa hemd wie immer[0m │[37m [0m[33m     [0m │[37m [0m[32m                   [0m │[37m [0m[32m     [0m │[37m [0m[32m      [0m │[37m [0m[32m     3[0m │[37m [0m[32m     [0m │[37m [0m[32m    6[0m │[37m [0m[32m      7[0m │[37m [0m[32m      [0m │[37m [0m[32m                 [0m │
└─────────────────────┴───────┴─────────────────────┴───────┴────────┴────────┴───────┴───────┴─────────┴────────┴───────────────────┘"
`;
